src/quicklist.c

```diff
@@ -1295,15 +1443,25 @@ int quicklistIndex(const quicklist *quicklist, const long long idx,
 void quicklistRotate(quicklist *quicklist) {
     if (quicklist->count <= 1)
         return;
-
     /* First, get the tail entry */
-    unsigned char *p = ziplistIndex(quicklist->tail->zl, -1);
+    unsigned char *p = NULL;
     unsigned char *value, *tmp;
     long long longval;
-    unsigned int sz;
+    size_t sz;
     char longstr[32] = {0};
-    ziplistGet(p, &tmp, &sz, &longval);
 
+    if(unlikely(quicklist->tail->container == QUICKLIST_NODE_CONTAINER_NONE))
+    {
+        /* Add tail entry to head (must happen before tail is deleted). */
+        quicklistPushHead(quicklist, quicklist->tail->entry, quicklist->tail->sz);
+        /* Remove tail entry. */
+        quicklistDelIndex(quicklist, quicklist->tail, &p);
```

Comment:
Would it be better to move tail to head, without having to copy and free memory?